### PR TITLE
Development

### PR DESCRIPTION
**Pull request by faysalali**
1) complete customization of devise default login and sign up pages.
2) Completed the user sign up, log in and logout functionality
3) Add the edit profile functionality
4) worked on indentations and basic layout
5) create an admin role. yet to completed.

### CODE REVIEW

File: app/assets/stylesheets/homepage.scss [hunk 1]
```diff
@@ -6,4 +6,11 @@
     color: blue;
     font-size: 60px;
     margin-top: 40px;
-}
\ No newline at end of file
+}
+.center_div{
+    margin: 0 auto;
```
Comment (taimur-akhtar) [inline app/assets/stylesheets/homepage.scss:11]: spacing.

File: app/assets/stylesheets/homepage.scss [hunk 2]
```diff
@@ -6,4 +6,11 @@
...
+    width:60% /* value of your choice which suits your alignment */
+}
+.margin-bottom {
+margin-bottom:15px;
```
Comment (taimur-akhtar) [inline app/assets/stylesheets/homepage.scss:15]: spacing

File: app/controllers/homepage_controller.rb [hunk 3]
```diff
@@ -1,5 +1,4 @@
 class HomepageController < ApplicationController
-	def index
-		
-	end
+  def index
```
Comment (taimur-akhtar) [inline app/controllers/homepage_controller.rb:2]: spacing.

File: app/views/devise/registrations/edit.html.erb [hunk 4]
```diff
@@ -1,43 +1,59 @@
...
+      </div>
+            
+      <div class="field">
+        <div><%= f.check_box :is_female, {}, true %> <%= f.label :is_female, "Female" %> </div>
+        <div><%= f.check_box :is_female, {}, false %> <%= f.label :is_female, "Male" %> </div>
+      </div>
+            
+      <div class="field">
+        <div><%= f.label :date_of_birth %><br />
+        <%= f.date_select :date_of_birth, :start_year => 1900%></div>
```
Comment (taimur-akhtar) [inline app/views/devise/registrations/edit.html.erb:19]: use notation start_year: 1900.

File: app/views/devise/registrations/edit.html.erb [hunk 5]
```diff
@@ -1,43 +1,59 @@
...
+      </div>
+            
+      <div class="field">
+        <%= f.label :email %><br />
+        <%= f.email_field :email, autofocus: true, autocomplete: "email",class: 'form-control margin-bottom'%>
```
Comment (taimur-akhtar) [inline app/views/devise/registrations/edit.html.erb:24]: spacing issue.

File: app/views/devise/registrations/new.html.erb [hunk 6]
```diff
@@ -1,29 +1,48 @@
...
+    <div class="col-lg-3">
+      <h2>Sign up</h2>
+      <%= form_for(resource, as: resource_name, url: registration_path(resource_name)) do |f| %>
+      <%= devise_error_messages! %>
+      <div class="field">
+        <label><b><div><%= f.label :name %></b></label><br />
+        <%= f.text_field :name, autofocus: true,class: 'form-control margin-bottom' %>
+      </div>
+      
+      <div class="field">
```
Comment (taimur-akhtar) [inline app/views/devise/registrations/new.html.erb:12]: we can take common entities to a partial.

File: db/migrate/20180719062610_add_name_to_users.rb [hunk 7]
```diff
@@ -0,0 +1,7 @@
+class AddNameToUsers < ActiveRecord::Migration[5.1]
+  def change
+    add_column :users, :name, :string
+    add_column :users, :date_of_birth, :datetime
+    add_column :users, :is_female, :boolean, default: false
```
Comment (taimur-akhtar) [inline db/migrate/20180719062610_add_name_to_users.rb:5]: use gender column.

File: db/migrate/20180719114107_add_role_to_users.rb [hunk 8]
```diff
@@ -0,0 +1,5 @@
+class AddRoleToUsers < ActiveRecord::Migration[5.1]
+  def change
+    add_column :users, :role, :integer
```
Comment (taimur-akhtar) [inline db/migrate/20180719114107_add_role_to_users.rb:3]: use enum for role.

File: db/migrate/20180719120404_add_role_to_users.rb [hunk 9]
```diff
@@ -0,0 +1,5 @@
+class AddRoleToUsers < ActiveRecord::Migration[5.1]
+  def change
```
Comment (taimur-akhtar) [inline db/migrate/20180719120404_add_role_to_users.rb:2]: why do we have two migrations? for the same thing?

File: app/assets/stylesheets/homepage.scss (hunk 2 above)
Comment (Faysalali534) [reply app/assets/stylesheets/homepage.scss:15]: Done.

File: app/controllers/homepage_controller.rb (hunk 3 above)
Comment (Faysalali534) [reply app/controllers/homepage_controller.rb:2]: Done.

File: app/views/devise/registrations/edit.html.erb (hunk 4 above)
Comment (Faysalali534) [reply app/views/devise/registrations/edit.html.erb:19]: Done.

File: app/views/devise/registrations/edit.html.erb (hunk 5 above)
Comment (Faysalali534) [reply app/views/devise/registrations/edit.html.erb:24]: Done.

File: db/migrate/20180719062610_add_name_to_users.rb (hunk 7 above)
Comment (Faysalali534) [reply db/migrate/20180719062610_add_name_to_users.rb:5]: Done.

File: db/migrate/20180719114107_add_role_to_users.rb (hunk 8 above)
Comment (Faysalali534) [reply db/migrate/20180719114107_add_role_to_users.rb:3]: Done.

File: db/migrate/20180719120404_add_role_to_users.rb (hunk 9 above)
Comment (Faysalali534) [reply db/migrate/20180719120404_add_role_to_users.rb:2]: Deleted.

**Pull request merged**